### PR TITLE
Shilpi newcastle crc

### PR DESCRIPTION
**Pull request by shilpigoeldev**
CRC restructuring as per final new rules , finished QA with Francesco which he has signed off . Kindly review 🙏

### CODE REVIEW

File: lib/import/colorectal/providers/newcastle/newcastle_handler_colorectal.rb [hunk 1]
```diff
@@ -7,341 +7,230 @@ module Providers
...
+              process_variants(genocolorectal, variant)
+              genocolorectals.append(genocolorectal)
+            elsif gene.present? # for other status records
+              add_fs_negative_genes(gene, genocolorectal, genocolorectals, record)
+              genocolorectals.append(genocolorectal)
+            else # for other status null gene record
+              process_null_gene_rec(genocolorectal, genocolorectals)
             end
+
+            genocolorectals
```
Comment (ollietulloch) [inline lib/import/colorectal/providers/newcastle/newcastle_handler_colorectal.rb:133]: @shilpigoeldev not sure you need to return `genocolorectals` here. The result of this method isn't assigned anywhere, the method mutates `genocolorectals`

File: lib/import/colorectal/providers/newcastle/newcastle_handler_colorectal.rb [hunk 2]
```diff
@@ -7,341 +7,230 @@ module Providers
...
+            return false if genocolorectal.attribute_map['genetictestscope'].nil?
+
+            genocolorectal.attribute_map['genetictestscope'].scan(/Unable/i).size.positive?
           end
+
+          def pathogenic?(record)
+            varpathclass = record.raw_fields['variantpathclass']&.downcase
+            return true if NON_PATHEGENIC_CODES.exclude? varpathclass
+
+            false
```
Comment (ollietulloch) [inline lib/import/colorectal/providers/newcastle/newcastle_handler_colorectal.rb:201]: HI @shilpigoeldev, no need to explicitly return `true` or `false` here, just have `NON_PATHEGENIC_CODES.exclude? varpathclass` be the return value of the method

File: lib/import/colorectal/providers/newcastle/newcastle_handler_colorectal.rb [hunk 3]
```diff
@@ -7,341 +7,230 @@ module Providers
...
+
+          def process_protein_impact(genocolorectal, variant)
+            if variant.scan(PROTEIN_REGEX).size.positive?
+              genocolorectal.add_protein_impact($LAST_MATCH_INFO[:impact])
+              @logger.debug "SUCCESSFUL protein parse for: #{$LAST_MATCH_INFO[:impact]}"
+            else
+              @logger.debug "FAILED protein parse for: #{variant}"
+            end
+          end
+
```
Comment (ollietulloch) [inline lib/import/colorectal/providers/newcastle/newcastle_handler_colorectal.rb:233]: Remove the blank line please :)

**Review by ollietulloch — CHANGES_REQUESTED** [review]
Hi @shilpigoeldev,

Just a couple of minor tweaks to make.

Thanks,
Ollie

File: lib/import/helpers/colorectal/providers/rtd/rtd_constants.rb [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+                                ([0-9]+[+>_-][0-9][+>_-][0-9]+[+>_-][0-9]+[ACGTdelinsup]+)|
+                                ([0-9]+[+>_-][0-9]+[ACGTdelinsup][+>_-][ACGTdelinsup])|
+                                ([0-9]+[ACGTdelinsup]+[+>_-][ACGTdelinsup])|
+                                ([0-9]+[+>_-][0-9]+[ACGTdelinsup]+)|
+                                ([0-9]+[+>_-][0-9]+[ACGTdelinsup]?)|
+                                ([0-9]+[+>_-][0-9]+[+>_-][0-9]+[0-9]+[ACGTdelinsup]+)|
+                                ([0-9]+[?+>_-]+[0-9]+[?+>_-]+[ACGTdelinsup]+)|
+                                ([0-9]+[ACGTdelinsup]+)|
+                                ([+>_-][0-9]+[ACGTdelinsup>]+)
+                                )\]?/ix.freeze
```
Comment (ollietulloch) [inline lib/import/helpers/colorectal/providers/rtd/rtd_constants.rb:76]: As of ruby 3.x, Rubocop will compain here:`Do not freeze immutable objects, as freezing them has no effect`

File: lib/import/colorectal/providers/newcastle/newcastle_handler_colorectal.rb (hunk 1 above)
Comment (shilpigoeldev) [reply lib/import/colorectal/providers/newcastle/newcastle_handler_colorectal.rb]: Done , on all the methods where not assigning to another variable.

File: lib/import/colorectal/providers/newcastle/newcastle_handler_colorectal.rb (hunk 2 above)
Comment (shilpigoeldev) [reply lib/import/colorectal/providers/newcastle/newcastle_handler_colorectal.rb]: done ✅

File: lib/import/colorectal/providers/newcastle/newcastle_handler_colorectal.rb (hunk 3 above)
Comment (shilpigoeldev) [reply lib/import/colorectal/providers/newcastle/newcastle_handler_colorectal.rb]: Removed it and rubocopped the whole file.

File: lib/import/helpers/colorectal/providers/rtd/rtd_constants.rb (hunk 4 above)
Comment (shilpigoeldev) [reply lib/import/helpers/colorectal/providers/rtd/rtd_constants.rb]: done ✅

**Comment by shilpigoeldev** [comment]
Thanks Ollie, I have done the changes and confirmed back the counts.

**Review by ollietulloch — APPROVED** [review]
Thanks @shilpigoeldev

There is one outstanding rubocop warning for the `NewcastleHandlerColorectal` class being too long, it'd be good to address that witha general boyscout PR at somepoint.

**Comment by shilpigoeldev** [comment]
Merged ✅

**Pull request merged**